### ISSUE TITLE
Node topology labels are not applied to PV backed OSDs on rook 1.2.0 after upgrade from v1.1

### ISSUE THREAD

**Issue by paalkr** [rook/rook#4616]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
When upgrading from rook 1.1.7 to 1.2.0 the node lables for region and zone described in https://github.com/rook/rook/blob/release-1.2/Documentation/ceph-cluster-crd.md#osd-topology
```
failure-domain.beta.kubernetes.io/region
failure-domain.beta.kubernetes.io/zone
```
are not applied to the OSD, causing the crush map to change during the upgrade.
![image](https://user-images.githubusercontent.com/9726307/71806899-55a89680-306a-11ea-8a24-e934df5d4f67.png)

**Expected behavior:**
The crush map to stay unchanged and equal to how it was before upgrading to 1.2.0
![image](https://user-images.githubusercontent.com/9726307/71806870-3d387c00-306a-11ea-981e-ed3857c0393b.png)

**How to reproduce it (minimal and precise):**
Deploy a 1.1.7 PV backed rook ceph cluster in AWS (or any other cluster that has the mentioned failure-domain annotations)

* Node labels (example from node in zone c)
```
Name:               ip-10-1-45-17.eu-west-1.compute.internal
Roles:              node,storageC
Labels:             beta.kubernetes.io/arch=amd64
                    beta.kubernetes.io/instance-type=m5a.large
                    beta.kubernetes.io/os=linux
                    cputype=normal
                    failure-domain.beta.kubernetes.io/region=eu-west-1
                    failure-domain.beta.kubernetes.io/zone=eu-west-1c
                    kube-aws.coreos.com/autoscalinggroup=k8s5-StorageC-1P1VQPZET13L8-Workers-1VWKDONC1ELQW
                    kube-aws.coreos.com/cluster=k8s5
                    kube-aws.coreos.com/launchconfiguration=
                    kube-aws.coreos.com/role=storage
                    kube-aws.coreos.com/zone=c
                    kubernetes.io/arch=amd64
                    kubernetes.io/hostname=ip-10-1-45-17.eu-west-1.compute.internal
                    kubernetes.io/os=linux
                    kubernetes.io/role=node
                    node-role.kubernetes.io/node=
                    node-role.kubernetes.io/storageC=
                    node.kubernetes.io/role=storageC
```

* Cluster CR (custom resource), typically called `cluster.yaml`, if necessary
```
apiVersion: ceph.rook.io/v1
kind: CephCluster
metadata:
  name: rook-ceph
  namespace: rook-ceph
spec:
  dataDirHostPath: /var/lib/rook/cluster
  mon:
    count: 3
    allowMultiplePerNode: false
    volumeClaimTemplate:
      spec:
        storageClassName: ebs-gp2
        resources:
          requests:
            storage: 10Gi
  mgr:
    modules:
    - name: pg_autoscaler
      enabled: true           
  placement:
    mon:
      nodeAffinity:
        requiredDuringSchedulingIgnoredDuringExecution:
          nodeSelectorTerms:
          - matchExpressions:
            - key: kube-aws.coreos.com/role
              operator: In
              values:
              - storage 
      # Pod Anti Affinity for the mon pods are handled through the spec.mon.allowMultiplePerNodesetting
      # podAntiAffinity:
        # requiredDuringSchedulingIgnoredDuringExecution:
        # - labelSelector:
            # matchExpressions:
            # - key: app
              # operator: In
              # values:
              # - rook-ceph-mon
          # topologyKey: failure-domain.beta.kubernetes.io/zone              
      tolerations:
        - key: kube-aws.coreos.com/role
          operator: Equal
          value: storage
          effect: NoSchedule 
    # mgr:
    #   nodeAffinity:
    #     requiredDuringSchedulingIgnoredDuringExecution:
    #       nodeSelectorTerms:
    #       - matchExpressions:
    #         - key: kube-aws.coreos.com/role
    #           operator: In
    #           values:
    #           - storage 
    #   tolerations:
    #     - key: kube-aws.coreos.com/role
    #       operator: Equal
    #       value: storage
    #       effect: NoSchedule               
  cephVersion:
    image: ceph/ceph:v14.2.4-20190917
    allowUnsupported: false
  dashboard:
    enabled: true
    urlPrefix: /ceph-dashboard
    port: 8443
    ssl: false
  monitoring:
    # requires Prometheus to be pre-installed
    enabled: true
    # namespace to deploy prometheusRule in. If empty, namespace of the cluster will be used.
    # Recommended:
    # If you have a single rook-ceph cluster, set the rulesNamespace to the same namespace as the cluster or keep it empty.
    # If you have multiple rook-ceph clusters in the same k8s cluster, choose the same namespace (ideally, namespace with prometheus
    # deployed) to set rulesNamespace for all the clusters. Otherwise, you will get duplicate alerts with multiple alert definitions.
    rulesNamespace: rook-ceph    
  network:
    hostNetwork: false
  storage:
    topologyAware: true
    storageClassDeviceSets:
    - name: zone-a
      count: 1
      portable: true
      resources:
        limits:
          cpu: "2000m"
          memory: "4Gi"
        requests:
          cpu: "500m"
          memory: "4Gi"
      placement:
        nodeAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
            nodeSelectorTerms:
            - matchExpressions:
              - key: kube-aws.coreos.com/zone
                operator: In
                values:
                - a 
              - key: kube-aws.coreos.com/role
                operator: In
                values:
                - storage   
        tolerations:
        - key: kube-aws.coreos.com/role
          operator: Equal
          value: storage
          effect: NoSchedule                
      volumeClaimTemplates:
      - metadata:
          name: data
        spec:
          resources:
            requests:
              storage: 250Gi
          storageClassName: ebs-gp2
          volumeMode: Block
          accessModes:
            - ReadWriteOnce
    - name: zone-b
      count: 1
      portable: true
      resources:
        limits:
          cpu: "2000m"
          memory: "4Gi"
        requests:
          cpu: "500m"
          memory: "4Gi"
      placement:
        nodeAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
            nodeSelectorTerms:
            - matchExpressions:
              - key: kube-aws.coreos.com/zone
                operator: In
                values:
                - b
              - key: kube-aws.coreos.com/role
                operator: In
                values:
                - storage  
        tolerations:
        - key: kube-aws.coreos.com/role
          operator: Equal
          value: storage
          effect: NoSchedule                
      volumeClaimTemplates:
      - metadata:
          name: data
        spec:
          resources:
            requests:
              storage: 250Gi
          storageClassName: ebs-gp2
          volumeMode: Block
          accessModes:
            - ReadWriteOnce 
    - name: zone-c
      count: 1
      portable: true
      resources:
        limits:
          cpu: "2000m"
          memory: "4Gi"
        requests:
          cpu: "500m"
          memory: "4Gi"
      placement:
        nodeAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
            nodeSelectorTerms:
            - matchExpressions:
              - key: kube-aws.coreos.com/zone
                operator: In
                values:
                - c
              - key: kube-aws.coreos.com/role
                operator: In
                values:
                - storage 
        tolerations:
        - key: kube-aws.coreos.com/role
          operator: Equal
          value: storage
          effect: NoSchedule
      volumeClaimTemplates:
      - metadata:
          name: data
        spec:
          resources:
            requests:
              storage: 250Gi
          storageClassName: ebs-gp2
          volumeMode: Block
          accessModes:
            - ReadWriteOnce     
```

**Environment**:
* OS (e.g. from /etc/os-release):
```
NAME="Container Linux by CoreOS"
ID=coreos
VERSION=2247.7.0
VERSION_ID=2247.7.0
BUILD_ID=2019-11-19-2251
PRETTY_NAME="Container Linux by CoreOS 2247.7.0 (Rhyolite)"
ANSI_COLOR="38;5;75"
HOME_URL="https://coreos.com/"
BUG_REPORT_URL="https://issues.coreos.com"
COREOS_BOARD="amd64-usr"
```
* Kernel (e.g. `uname -a`):
`Linux ip-10-1-45-17.eu-west-1.compute.internal 4.19.84-coreos #1 SMP Tue Nov 19 20:51:06 -00 2019 x86_64 AMD EPYC 7571 AuthenticAMD GNU/Linux`
* Cloud provider or hardware configuration:
AWS
* Rook version (use `rook version` inside of a Rook Pod):
upgrade from 1.1.7 to 1.2.0
* Storage backend version (e.g. for ceph do `ceph -v`):
v14.2.4-20190917
* Kubernetes version (use `kubectl version`):
1.14.8
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):
Self managed K8s cluster deployed to AWS using the [kube-aws tool](https://github.com/kubernetes-incubator/kube-aws)
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox]
Before upgrade
```
cluster:
    id:     cd74ad6e-183f-4593-8e0c-717e61fee519
    health: HEALTH_OK
  services:
    mon: 3 daemons, quorum a,b,c (age 3w)
    mgr: a(active, since 12d)
    mds: r2-fs:3 {0=r2-fs-b=up:active,1=r2-fs-c=up:active,2=r2-fs-e=up:active} 3 up:standby
    osd: 3 osds: 3 up (since 3w), 3 in (since 3w)
  data:
    pools:   4 pools, 32 pgs
    objects: 3.74k objects, 13 GiB
    usage:   24 GiB used, 723 GiB / 747 GiB avail
    pgs:     32 active+clean
  io:
    client:   170 B/s rd, 10 KiB/s wr, 0 op/s rd, 1 op/s wr
```

After upgrade
```
cluster:
    id:     cd74ad6e-183f-4593-8e0c-717e61fee519
    health: HEALTH_WARN
            1 filesystem is degraded
            3 MDSs report slow metadata IOs
            Reduced data availability: 23 pgs inactive, 3 pgs peering, 17 pgs stale
            Degraded data redundancy: 865/10877 objects degraded (7.953%), 14 pgs degraded, 16 pgs undersized
            5 pgs not deep-scrubbed in time
            5 pgs not scrubbed in time
  services:
    mon: 3 daemons, quorum a,b,c (age 17m)
    mgr: a(active, since 16m)
    mds: r2-fs:3/3 {0=r2-fs-e=up:replay,1=r2-fs-d=up:replay,2=r2-fs-c=up:replay} 3 up:standby
    osd: 3 osds: 3 up (since 15m), 3 in (since 3w); 7 remapped pgs
  data:
    pools:   4 pools, 32 pgs
    objects: 3.65k objects, 13 GiB
    usage:   24 GiB used, 723 GiB / 747 GiB avail
    pgs:     15.625% pgs unknown
             56.250% pgs not active
             865/10877 objects degraded (7.953%)
             6771/10877 objects misplaced (62.251%)
             11 stale+undersized+degraded+peered
             7  active+clean+remapped
             5  unknown
             3  stale+peering
             2  undersized+degraded+peered
             1  stale+active+undersized+degraded+remapped
             1  stale+active+clean
             1  stale+undersized+peered
             1  undersized+peered
```

**Comment by paalkr** [comment]
Probably the same issue as reported in https://github.com/rook/rook/issues/4543

**Comment by travisn** [comment]
Looks like #4543 only fixed this issue for the scenario of restarting the operator, assuming you had initially deployed a v1.2 cluster. If upgraded from v1.1, the fix for #4543 does not apply. If the `--crush-location` property is not found on the pod spec, the operator will need to determine the value based on the node labels instead of skipping this setting.

**Comment by paalkr** [comment]
Will both this issue and the issue addressed in https://github.com/rook/rook/issues/4543 be targeted at the 1.2.2 release?

**Comment by travisn** [comment]
@paalkr Yes I would expect this to be fixed in v1.2.2. The issue fixed by #4543 was already included in v1.2.0.

**Comment by paalkr** [comment]
I can confirm that upgrading does work as expected with the 1.2.2 release. Thanks!
